wwwapp/models.py

```diff
@@ -59,6 +59,47 @@ class WorkshopUserProfile(models.Model):
     def __str__(self):
         return '%s: %s, %s' % (self.year, self.user_profile, self.status)
 
+class PESELField(models.CharField):
+    """PESEL field, with checksum verification."""
+
+    def validate(self, pesel: str, user_info: 'UserInfo') -> None:
```

Comment:
I can't find documentation for this method. Is this method even being called by Django?

Reply:
Yes it does, although the docs specify that it should be used only for `validation that is not suitable for a validator.` (i.e. consider using a validator instead) - although in this specific case I think it makes sense (it's unlikely you'll want to reuse this check on anything other than PESELField and a simple CharField won't do because we need additional functions for extracting the date)
https://docs.djangoproject.com/en/1.11/ref/forms/validation/